src/Corax/IndexEntry/IndexEntryWriter.cs

```diff
@@ -105,13 +105,128 @@ public void Write(int field, ReadOnlySpan<byte> value)
             _dataIndex += sizeof(IndexEntryFieldType);
             return;
         }
-        
+
         int length = VariableSizeEncoding.Write(buffer, value.Length, _dataIndex);
         _dataIndex += length;
         value.CopyTo(buffer.Slice(_dataIndex, value.Length));
         _dataIndex += value.Length;
     }
 
+    public void WriteNullDynamic(string name)
+    {
+        using var _ = _context.From(name, ByteStringType.Immutable, out var fieldNameStr);
+        Span<byte> tmpBuffer = stackalloc byte[10];
+        int fieldLenLen = VariableSizeEncoding.Write(tmpBuffer, fieldNameStr.Length);
```

Comment:
Why are we writing this into a temporary buffer?

This `stackalloc` would cause the following sequence of instructions. 
```
push 2
push 2
push 2
push 2
push 2
```
plus we are performing an encoding of the size when we are gonna discard the `tmpBuffer` later. Instead just do a guestimation of the biggest value and be done with it. 
```
var requiredSize = fieldLenLen + 2 * sizeof(long);
```

Reply:
I'll fix that